Sources/LanguageServerProtocolJSONRPC/MessageCoding.swift

```diff
@@ -17,7 +17,7 @@ public enum JSONRPCMessage {
   case notification(NotificationType)
   case request(_RequestType, id: RequestID)
   case response(ResponseType, id: RequestID)
-  case errorResponse(ResponseError, id: RequestID)
```

Comment:
We decode this value below like so:

```swift
      case (let id?, nil, _, let error?):
        msgKind = .response
        self = .errorResponse(error, id: id)
```

... which assumes that the `id` is non-nil.  That probably needs to get updated to `let id` instead of `let id?`.  Please add a test to `Tests/LanguageServerProtocolJSONRPCTests/CodingTests.swift` ensuring that we can round-trip this correctly when `id == nil`.

Reply:
done